tests/src/task.rs

```diff
@@ -0,0 +1,112 @@
+use crate::common;
+use aerospike::*;
+use aerospike::task::{Status, Task};
+use aerospike::errors::{ErrorKind};
+use std::{thread};
+use std::time::{Duration};
+
+
+// TODO: replace matches_override with matches when upgrade to 1.42.0
+#[macro_export]
+macro_rules! matches_override {
+    ($expression:expr, $($pattern:tt)+) => {
+        match $expression {
+            $($pattern)+ => true,
+            _ => false
+        }
+    }
+}
+
+// If registering udf is successful, querying RegisterTask will return Status::Complete 
+// If udf does not exist, querying RegisterTask will return error
+#[test]
+fn register_task_test() {
+    let client = common::client();
+
+    let code = r#"
+    local function putBin(r,name,value)
+        if not aerospike:exists(r) then aerospike:create(r) end
+        r[name] = value
+        aerospike:update(r)
+    end
+    function writeBin(r,name,value)
+        putBin(r,name,value)
+    end
+    "#;
+
+
+    let udf_name = common::rand_str(10);
+    let udf_file_name = udf_name.clone().to_owned() + ".LUA";
+
+    let register_task = client.register_udf(
+        &WritePolicy::default(),
+        code.as_bytes(),
+        &udf_file_name,
+        UDFLang::Lua).unwrap();
+
+    assert!(matches_override!(register_task.wait_till_complete().unwrap(), Status::Complete));
+
+    client.remove_udf(
+        &WritePolicy::default(),
+        &udf_name,
+        UDFLang::Lua
+    ).unwrap();
+
+
+    thread::sleep(Duration::from_secs(5));
+
+    match register_task.wait_till_complete() {
+        Err(e) => {
+            assert!(matches_override!(e.kind(), ErrorKind::Connection(_err_string)));
+        },
+        _ => {
+            panic!("error");
+        }
+
+    }
```

Comment:
Can simplify this a bit by using `matches` on the output of `wait_till_complete` directly:

```suggestion
    assert!(matches_override!(
        register_task.wait_till_complete(),
        Err(Error(ErrorKind::Connection(_), _))
    ));
```